NitroxServer/Communication/Packets/PacketHandler.cs

```diff
@@ -12,26 +13,28 @@ namespace NitroxServer.Communication.Packets
     public class PacketHandler
     {
         private readonly PlayerManager playerManager;
+        private readonly JoiningManager joiningManager;
         private readonly DefaultServerPacketProcessor defaultServerPacketProcessor;
         private readonly Dictionary<Type, PacketProcessor> packetProcessorAuthCache = new();
         private readonly Dictionary<Type, PacketProcessor> packetProcessorUnauthCache = new();
 
-        public PacketHandler(PlayerManager playerManager, DefaultServerPacketProcessor packetProcessor)
+        public PacketHandler(PlayerManager playerManager, JoiningManager joiningManager, DefaultServerPacketProcessor packetProcessor)
         {
             this.playerManager = playerManager;
+            this.joiningManager = joiningManager;
             defaultServerPacketProcessor = packetProcessor;
         }
 
         public void Process(Packet packet, INitroxConnection connection)
         {
             Player player = playerManager.GetPlayer(connection);
-            if (player == null)
+            if (player != null)
             {
-                ProcessUnauthenticated(packet, connection);
+                ProcessAuthenticated(packet, player);
             }
-            else
+            else if (!joiningManager.GetQueuedPlayers().Contains(connection))
```

Comment:
Isn't it too harsh on the connection ? o(n) on packet receiving (a lot per second) might be a load too bad for the server

Reply:
I highly doubt this could cause performance problems since it only runs for unauthenticated packets (sent very infrequently compared to normal packets) and there will realistically never be more than a few people in the queue at once. It's best to use a profiler to confirm that a piece of code is actually causing a bottleneck before trying to optimize it.